### ISSUE TITLE
Reduce code duplication in workflows: Construct Input schema to workflows one level higher 

### ISSUE THREAD

**Issue by hmacdope** [choderalab/asapdiscovery#607]
At dev sync today we had a very fruitful discussion with @kaminow , @apayne97  and @ijpulidos  about how to avoid the following kind of duplication where a parameter goes from:

CLI -> unpacked into Inputs -> unpacked into Workflow -> unpacked into class in Workflow

```python
# cli 
def cli(blah=x):
   inputs = Inputs(blah=blah)
   
   wf(inputs)
  
# workflow
def wf(inputs):
     # use blah
     Blah(blah=inputs.blah)
 ```
 
 Where instead the `Blah` class (as they are factories) can just be instantiated directly at the CLI layer.

```python
# cli 
def cli(blah=x):
   inputs = Inputs(blah=Blah(blah=blah) # embed instantiated model directly (may not be possible for all models)
   wf(inputs)
  
# workflow
def wf(inputs):
     # use blah
     inputs.blah() # use instantiated model
 ```

I propose we make this change after #594 and #595 as will be a bit of a change of design.

**Comment by apayne97** [comment]
Does this new way of doing it make this PR obsolete? #596

**Comment by hmacdope** [comment]
Yes but I think we would need to do a test run to see how easy to implement. I'll put it on roadmap?

**Comment by apayne97** [comment]
How easy is it to serialize nested schema? That's one concern I had - I feel like I've struggled with that before but maybe we're doing it in a smarter way this time

**Comment by kaminow** [comment]
I think pydantic should handle it natively, I don't think we need to do anything special
